### DISCUSSION TITLE
Is it possible to render MiniMap outside of ReactFlow

### DISCUSSION THREAD

**Discussion by goliney** [xyflow/xyflow#3156] [Q&A] [answered]
I want to use [MiniMap](https://reactflow.dev/docs/api/plugin-components/minimap/), but sometimes it occupies valuable space. When there are a lot of nodes the minimap may cover important content. It becomes problematic for my project.

I have a sidebar that is rendered outside the `<ReactFlow />` container. Is there a way to move `<MiniMap />` there, outside of `<ReactFlow />`? Perhaps, using portals 🤔

**Comment by moklick** [comment, marked as answer]
You can render the Minimap and the Controls (probably also the Background :D) outside of the ReactFlow component if you wrap your application with the ReactFlowProvider: https://codesandbox.io/s/wandering-worker-vn9x2m?file=/src/App.tsx:172-192
